### PR TITLE
Genesis/athens3 reduced steps

### PR DESCRIPTION
**Pull request by kingpinXD**

### CODE REVIEW

File: docs/steps.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+sudo apt update
+sudo apt install -y jq
+jq --version
+```
+
+```bash
+git clone https://github.com/zeta-chain/network-athens3.git && cd network-athens3
+```
+
+- Branch name will be provided by the coordinator
```
Comment (CharlieMc0) [inline docs/steps.md:13]: Who is the coordinator? Looking at this from the outside operator perspective they won't have any context for who/what the coordinator is.  

I think changing this to  "zetachain coordinator" will make it a little clearer to outside operators

File: network_files/config/app.toml [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+rpc-read-timeout = 10
+
+# RPCWriteTimeout defines the Tendermint RPC write timeout (in seconds).
+rpc-write-timeout = 0
+
+# RPCMaxBodyBytes defines the Tendermint maximum response body (in bytes).
+rpc-max-body-bytes = 1000000
+
+# EnableUnsafeCORS defines if CORS should be enabled (unsafe - use it at your own risk).
+enabled-unsafe-cors = true
```
Comment (CharlieMc0) [inline network_files/config/app.toml:133]: FALSE - Can't be recommending operators run have this enabled

File: network_files/config/app.toml [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+]
+
+###############################################################################
+###                           API Configuration                             ###
+###############################################################################
+
+[api]
+
+# Enable defines if the API server should be enabled.
+enable = true
```
Comment (CharlieMc0) [inline network_files/config/app.toml:112]: Is this required for validator nodes? If not, it should be set to false

File: network_files/config/app.toml [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+max-tx-gas-wanted = 0
+
+###############################################################################
+###                           JSON RPC Configuration                        ###
+###############################################################################
+
+[json-rpc]
+
+# Enable defines if the gRPC server should be enabled.
+enable = true
```
Comment (CharlieMc0) [inline network_files/config/app.toml:223]: FALSE - Can't be recommending operators run have this enabled

File: network_files/config/app.toml [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+
+# Address defines the EVM RPC HTTP server address to bind to.
+address = "0.0.0.0:8545"
```
Comment (CharlieMc0) [inline network_files/config/app.toml:226]: Should be `127.0.0.1` by default

File: network_files/config/app.toml [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+
+# Address defines the EVM WebSocket server address to bind to.
+ws-address = "0.0.0.0:8546"
```
Comment (CharlieMc0) [inline network_files/config/app.toml:229]: Should be `127.0.0.1` by default

File: network_files/config/client.toml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+# For more information, see https://github.com/toml-lang/toml
+
+###############################################################################
+###                           Client Configuration                            ###
+###############################################################################
+
+# The network chain ID
+chain-id = ""
+# The keyring's backend, where the keys are stored (os|file|kwallet|pass|test|memory)
+keyring-backend = "test"
```
Comment (CharlieMc0) [inline network_files/config/client.toml:11]: KEYRING should not be test. We should be using and encouraging all operators to use OS, kwallet, or maybe file backends

**Review by CharlieMc0 — COMMENTED** [review]
Left comments about some unclear doc steps and insecure default configs we should not be recommending to operators

File: scripts/start-zetacore.sh [hunk 8]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+CHAINID="athens_7001-1"
+KEYRING="test"
+HOSTNAME=$(hostname)
+
+rm -rf ~/.zetacored/config/genesis.json
+cp -a network_files/config/genesis.json ~/.zetacored/config/
+zetacored start --pruning=nothing --minimum-gas-prices=0.0001azeta --json-rpc.api eth,txpool,personal,net,debug,web3,miner --api.enable >> ~/.zetacored/zetacored.log 2>&1  &
```
[the remark this answers is not in the document]
Comment (brewmaster012) [reply scripts/start-zetacore.sh:8]: right now zetaclient does not support keyring backends other than test.

File: scripts/node-setup.sh [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+rm -rf ~/.zetacored
+# create keys
+CHAINID="athens_7001-1"
+KEYRING="test"
+HOSTNAME=$(hostname)
+echo "HOSTNAME: $HOSTNAME"
+
+# Init a new node to generate genesis file .
+# Copy config files from existing folders which get copied via Docker Copy when building images
+zetacored init Zetanode_HOSTNAME --chain-id=$CHAINID
```
[the remark this answers is not in the document]
Comment (brewmaster012) [reply scripts/node-setup.sh:15]: this field should be set by val operator.

**Comment by CharlieMc0** [comment]
What about steps to generate and delegate keys?
Node setup doesn't specific supported OS's

File: scripts/node-setup.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (CharlieMc0) [reply scripts/node-setup.sh:15]: You're saying `Zetanode_HOSTNAME` should be replaced by the val operator? Can that be more obvious like `<REPLACE_HOSTNAME_HERE>`

**Comment by brewmaster012** [comment]
@CharlieMc0 @kingpinXD we have a few things that must be set by the val operators. 

1. SEEDIP: (we will need to tell them a SEEDIP, supposed one of our node)
2. moniker: their own choice

let's use environment variable for these things?

File: scripts/node-setup.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (kingpinXD) [reply scripts/node-setup.sh:15]: This doesn't matter, the config file gets replaced. 
Check lines 30 -32

**Comment by brewmaster012** [comment]
added two docs from notion--need to be updated as they might contain out-dated or conflicting information. 
@CharlieMc0 @kingpinXD

**Comment by kingpinXD** [comment]
> @CharlieMc0 @kingpinXD we have a few things that must be set by the val operators.
> 
> 1. SEEDIP: (we will need to tell them a SEEDIP, supposed one of our node)
> 2. moniker: their own choice
> 
> let's use environment variable for these things?

Moniker is already set by the script right now , it uses Zetacore_Hostname. We can maybe just a step in the docs , to allow them to change that. 
Or it can be a argument to the node-setup script .  

For SEED IP , Persistant peers etc We can probably set do the following 
- Set up a couple of our nodes first 
- Make changes to config.toml for persistent peers 
- Make changes to the start-zetaclient.sh for SEEDIP . 

Then circulate the files and steps to external nodes .

**Comment by CharlieMc0** [comment]
Everything here references zetacored -- Are we not recommending cosmovisor anymore?

File: docs/steps.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+**Wait until zetachain coordinator confirms that genesis and TSS keygen is completed**. 
+Then terminate the processes `zetacored` and `zetaclientd`.
+
+```bash
+pkill zetaclientd
+pkill zetacored
+```
+
+
+## Setup Process Management for `zetacored` and `zetaclientd`
```
Comment (brewmaster012) [inline docs/steps.md:78]: @CharlieMc0

File: docs/steps.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+NOTE : A backup us created for the existing zetacored folder under `~/.zetacored_old/zetacored-<timestamp>`.You can copy back keys etc if needed .
+
+  
+
+### Start Node Step
+
+**Wait for final genesis to be provided by the Coordinator of the ceremony before starting the following process**
+
+
+Edit config file (~/.zetacored/config/config.toml) to
```
Comment (brewmaster012) [inline docs/steps.md:53]: @kingpinXD i think we need to git pull here get the latest genesis.json file right?

File: docs/steps.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+sudo apt install -y jq curl git 
+jq --version
+```
+
+Now clone the network github repository
+```bash
+git clone https://github.com/zeta-chain/network-athens3.git && cd network-athens3
+```
+
+Get two executable binaries (`zetacored` and `zetaclientd`)
```
Comment (brewmaster012) [inline docs/steps.md:18]: @CharlieMc0

File: docs/steps.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
+### Node Setup
+Here we assume a typical Ubuntu 22.04 LTS x86_64 setup. If you are using a different OS you may need to make some adjustments.
+For more information about the compute node requirement see [here](hosting.md)
```
Comment (brewmaster012) [inline docs/steps.md:3]: @CharlieMc0

File: docs/steps.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+Get two executable binaries (`zetacored` and `zetaclientd`)
+- [ ] Add instructions here
+- [ ] Copy the binaries to `/usr/bin/` folder
+
+Check out a specific branch. Branch name will be provided by the coordinator
+```bash
+git checkout <Branch-Name>
+```
```
Comment (CharlieMc0) [inline docs/steps.md:25]: Why not use main? That's one less step for them

File: scripts/node-setup.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (CharlieMc0) [reply scripts/node-setup.sh:15]: I guess I'm unclear of when step would be ran and when the node-setup.sh script is ran.

If the moniker is pulled from the gentx file in lines 30-32 but you can't create the gentx file until after it has been init and the operator keys configured, when do you run this setup script?

File: docs/steps.md (hunk 11 above)
Comment (brewmaster012) [reply docs/steps.md:57]: ^^ @kingpinXD

File: docs/steps.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+sudo apt install -y jq curl git 
+jq --version
+```
+
+#### Download and install `zetacored` and `zetaclientd` binaries
+```
+wget https://zetachain-external-files.s3.amazonaws.com/binaries/athens3/latest/zetacored-ubuntu
+wget https://zetachain-external-files.s3.amazonaws.com/binaries/athens3/latest/zetaclientd-ubuntu
+mv zetaclientd-ubuntu /usr/bin/zetacored && chmod +x /usr/bin/zetacored
+mv zetaclientd-ubuntu /usr/bin/zetaclientd && chmod +x zetaclientd
```
Comment (brewmaster012) [inline docs/steps.md:18]: same as above @CharlieMc0

File: docs/steps.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+```bash
+chmod +x ./scripts/*.sh
+```
+
+```bash
+./scripts/node-setup.sh
+```
+
+- Create a branch , Commit , and Raise a PR
```
[the remark this answers is not in the document]
Comment (brewmaster012) [reply docs/steps.md]: yeah, can you figure this out and write a short doc for that?

File: docs/genesis.md [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+Your node must be setup as a validator and it will be part of the active set at its initial state. 
+
+### Deploy a Node
+
+If you have not already deployed a node, follow the instruction below but stop at **Install Genesis File.**
+
+[Hosting a Node](https://www.notion.so/Hosting-a-Node-64709ea14b6549b8abd45cd3299e8bff)
+
+In the following process, you will participate in generating the genesis file.
+
```
[the remark this answers is not in the document]
Comment (CharlieMc0) [reply docs/genesis.md:29]: @afzeta Agreed. I added a comment at the top of the doc. This doc is onlyneeded if they want to do it manually 

```
** These steps have been automated for the Athens3 testnet via the
`node-setup.sh` and `start-zetaclient.sh` scripts. Operators should ignore this
document unless you they to perform their key generation manually**
```

File: docs/steps.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (CharlieMc0) [reply docs/steps.md]: No access token should be required and don't recreate documentation for a PR that already exist. We just add a note `If you are unfamiliar with how to create and submit a PR please review the GitHub documentation` and link to https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

File: scripts/start-zetaclient.sh [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+echo "SEEDIP: $SEEDIP"
+echo "KeygenBlock: $KeygenBlock"
+export SEED=$(curl --retry 10 --retry-delay 5 --retry-connrefused  -s $SEEDIP:8123/p2p)
+export TSSPATH=~/.tss
+
+operatorAddress=$(zetacored keys show operator -a --keyring-backend=test)
+echo "operatorAddress: $operatorAddress"
+
+rm ~/.tss/*
+zetaclientd init --enable-chains "goerli_testnet,btc_testnet" \
```
Comment (CharlieMc0) [inline scripts/start-zetaclient.sh:43]: @kingpinXD 

Missing chains - What are the correct names for each chain? The examples in `zetaclientd --help` is wrong

`--enable-chains string   enable chains, comma separated list (default "GOERLI,BSCTESTNET,MUMBAI,ROPSTEN,BAOBAB")`

**Review by CharlieMc0 — APPROVED** [review]
Still needs work like support for core validators but fine to merge it now with the assumption it is still a WIP

**Pull request merged**